echopype/tests/convert/test_convert_api.py

```diff
@@ -0,0 +1,167 @@
+import os
+import pytest
+from typing import List, Tuple
+
+import xarray as xr
+
+from echopype.convert.api import open_raw
+
+
+@pytest.fixture
+def ek60_path(test_path):
+    return test_path["EK60"]
+
+
+def compare_zarr_vars(
+    ed_zarr: xr.Dataset, ed_no_zarr: xr.Dataset, var_to_comp: List[str], ed_path
+) -> Tuple[xr.Dataset, xr.Dataset]:
+    """
+    Compares the dask variables in ``ed_zarr`` against their
+    counterparts in ``ed_no_zarr`` by computing the dask results
+    and using xarray to make sure the variables are identical.
+    Additionally, this function will drop all of these compared
+    variables.
+
+    Parameters
+    ----------
+    ed_zarr : xr.Dataset
+        EchoData object with variables that were written directly
+        to a zarr and then loaded with dask
+    ed_no_zarr : xr.Dataset
+        An in-memory EchoData object
+    var_to_comp : List[str]
+        List representing those variables that were written directly
+        to a zarr and then loaded with dask
+    ed_path : str
+        EchoData group (e.g. "Sonar/Beam_group1")
+
+    Returns
+    -------
+    Tuple[xr.Dataset, xr.Dataset]
+        Datasets ``ed_zarr`` and ``ed_no_zarr``, respectively with
+        ``var_to_comp`` removed.
+    """
+
+    for var in var_to_comp:
+        for chan in ed_zarr[ed_path][var].channel:
+            # here we compute to make sure values are being compared, rather than just shapes
+            var_zarr = ed_zarr[ed_path][var].sel(channel=chan).compute()
+            var_no_zarr = ed_no_zarr[ed_path][var].sel(channel=chan)
+
+            assert var_zarr.identical(var_no_zarr)
+
+    ed_zarr[ed_path] = ed_zarr[ed_path].drop_vars(var_to_comp)
+    ed_no_zarr[ed_path] = ed_no_zarr[ed_path].drop_vars(var_to_comp)
+    return ed_zarr, ed_no_zarr
+
+
+@pytest.mark.parametrize(
+    ["raw_file", "sonar_model", "use_swap"],
+    [
+        ("L0003-D20040909-T161906-EK60.raw", "EK60", True),
+        pytest.param(
+            "L0003-D20040909-T161906-EK60.raw",
+            "EK60",
+            False,
+            marks=pytest.mark.xfail(
+                run=False,
+                reason="Expected out of memory error. See https://github.com/OSOceanAcoustics/echopype/issues/489",
+            ),
+        ),
+    ],
+    ids=["noaa_offloaded", "noaa_not_offloaded"],
```

Comment:
since now the argument is `use_swap`, maybe we change these too:
```suggestion
    ids=["noaa_use_swap_true", "noaa_use_swap_false"],
```